File: 6. semestr/[IBT] Bakalářská práce/Implementace/backend/Entity/Collision.cs
/*******************************************************************/
/*  Project: Bachelor's thesis – Genealogy Database System         */
/*           Faculty of Information Technology                     */
/*           Brno University of Technology                         */
/*                                                                 */
/*  File:    Collision.cs                                          */
/*  Date:    2019 – 2020                                           */
/*  Author:  Martin Chládek <xchlad16@stud.fit.vutbr.cz>           */
/*******************************************************************/

using Entity.Base;
using Entity.Enums;
using Entity.Relationships;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entity
{
    [Table("Collisions")]
    public class Collision : BaseEntity
    {
        public CollisionTypesEnum Type { get; set; }

        // References

        // Many to many - with Relationship
        public ICollection<CollisionRelationship> CollisionRelationship { get; set; }

        // Many to many - with FamilyTree
        public ICollection<FamilyTreeCollision> FamilyTreeCollision { get; set; }
    }
}

File: 6. semestr/[IBT] Bakalářská práce/Implementace/frontend/src/app/models/collision.ts
/*******************************************************************/
/*  Project: Bachelor's thesis – Genealogy Database System         */
/*           Faculty of Information Technology                     */
/*           Brno University of Technology                         */
/*                                                                 */
/*  File:    collision.ts                                          */
/*  Date:    2019 – 2020                                           */
/*  Author:  Martin Chládek <xchlad16@stud.fit.vutbr.cz>           */
/*******************************************************************/

import { CollisionRelationship } from './relationships/collisionRelationship';
import { FamilyTreeCollision } from './relationships/familyTreeCollisions';

export enum CollisionTypesEnum {
    differentMother,
    differentFather,
    marriageOrAncestor
}

export interface Collision {

    id: number;
    type: CollisionTypesEnum;

    // References

    // Many to many - with Relationship
    collisionRelationship: CollisionRelationship[];

    // Many to many - with FamilyTree
    familyTreeCollision: FamilyTreeCollision[];
}
